sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -836,6 +836,16 @@ case class RepartitionByExpression(
     child: LogicalPlan,
     numPartitions: Int) extends RepartitionOperation {
 
+  val (sortOrder, nonSortOrder) = partitionExpressions.partition(_.isInstanceOf[SortOrder])
+
+  require(sortOrder.isEmpty || nonSortOrder.isEmpty,
+    s"""${getClass.getSimpleName} expects that either all its `partitionExpressions` are of type
```

Comment:
Do you want this to be a multiline message? it makes sense to put the sort order and non sort order on new lines.